crates/sui-types/src/gas.rs

```diff
@@ -221,12 +221,28 @@ impl<'a> SuiGasStatus<'a> {
         computation_gas_unit_price: GasPrice,
         storage_gas_unit_price: GasPrice,
     ) -> SuiGasStatus<'a> {
-        Self::new(
-            GasStatus::new(&INITIAL_COST_SCHEDULE, GasUnits::new(gas_budget)),
+        let charge = true;
+        Self::new_with_budget_impl(
             gas_budget,
-            true,
             computation_gas_unit_price,
-            storage_gas_unit_price.into(),
+            storage_gas_unit_price,
+            charge,
+        )
+    }
+
+    /// WARNING! This should only be used in tests or in dev inspect and dry run. Gas will still
+    /// be measured, but not charged at the end of execution
+    pub fn new_uncharged(
+        gas_budget: u64,
+        computation_gas_unit_price: GasPrice,
+        storage_gas_unit_price: GasPrice,
+    ) -> SuiGasStatus<'a> {
+        let charge = false;
+        Self::new_with_budget_impl(
+            gas_budget,
+            computation_gas_unit_price,
+            storage_gas_unit_price,
+            charge,
         )
```

Comment:
I'm guessing this pattern is used to label the boolean literal with something more informative, is it common?  On first glance I struggled a bit to catch the difference this function and the one above vs the inline comment style, but I'm happy to get used to this style -- not a big deal.

```suggestion
        Self::new_with_budget_impl(
            gas_budget,
            computation_gas_unit_price,
            storage_gas_unit_price,
            /* charge */ false,
        )
```

Reply:
I've switched back and forth with both, so I'm not sure what is preferred. I sometimes like the variable name one only in that I find it easier to move around (and it sometimes is better for formatting) though I don't think that either case comes into play here